Libraries/PeriphDrivers/Source/UART/uart_reva.c

```diff
@@ -601,7 +620,6 @@ int MXC_UART_RevA_ReadRXFIFODMA(mxc_uart_reva_regs_t *uart, mxc_dma_regs_t *dma,
     srcdst.dest = bytes;
     srcdst.len = len;
 
-    states[uart_num].channelRx = channel;
     MXC_DMA_ConfigChannel(config, srcdst);
     MXC_DMA_SetCallback(channel, MXC_UART_DMACallback);
```

Comment:
should it not be callback instead of MXCUART_DMACallback ?

Reply:
@aniktash In this case we want to assign the callback to the driver's internal one, because it will check to ensure that both the RX and TX components of the transaction are completed before calling the user's callback.

`MXC_UART_DMACallback` eventually gets routed to `MXC_UART_RevA_DMACallback` - see [here](https://github.com/analogdevicesinc/msdk/blob/6cb136f68ca6c4fdac650a7d59d97550222bf29d/Libraries/PeriphDrivers/Source/UART/uart_reva.c#L1024)